classwork/сlasswork2.1/main.cpp

```diff
@@ -0,0 +1,108 @@
+#include <iostream>
+#include <unordered_map>
+#include <fstream>
+#include <string>
+#include <print>
+#include <format>
+
+namespace {
+    constexpr const char* DEFAULT_FILE_NAME = "user_stats.txt";
+    constexpr const char* RESET_COMMAND = "bread";
+    constexpr const char* DELETE_COMMAND = "delete";
+}
+
+class UserStatistics {
+public:
+    explicit UserStatistics(const std::string& statisticsFilePath = DEFAULT_FILE_NAME)
+            : fileName(statisticsFilePath) {
+        loadFromFile();
+    }
+
+    ~UserStatistics() {
+        saveToFile();
+    }
+
+    bool incrementCounterForUser(const std::string& name, std::string& message) {
+        if (name == RESET_COMMAND) {
+            userStats.clear();
+            saveToFile();
+            message = "All history has been exterminated.";
+            return true;
+        }
+
+        if (userStats.find(name) == userStats.end()) {
+            userStats[name] = 1;
+            message = std::format("Welcome, {}!", name);
```

Comment:
there are 2 comments here:
1. For both cases we can just use userStats[name]++ as if corresponding key was not in the map then it will be initialised with default value which is 0 in case of integral types.
2. In this function and in resetUser we have some parts of business logic like formatting of some user facing strings and that is violation of single responsibility principle - cause that strings belongs to data representation while other part of class is about data itself. Therefore it is better to cleanup this class from that message. for example this method can just return int - updated counter and based on that client of the class may implement different user facing scenario. In addition we will be able to use the same class for different programs without a need to rewrite big part of code. Data management class can be supported and changed independently from business logic e.g. scenarios of data usage